### ISSUE TITLE
bug: Build results in StackOverflowError

### ISSUE THREAD

**Issue by kokx** [icpc/live-v3#70]
## Context (Environment)
Trying to build the application using `./gradlew build` or running using `./gradlew run`.

## Expected Behavior
The build succeeds without an error, and its possible to run `./gradlew release` afterwards and use the resulting jar, or run the application.

## Current Behavior
When running `./gradlew run` or `./gradlew build`, the build results in a `java.lang.StackOverflowError`, failing the build. Full logs: https://gist.github.com/kokx/ca709aebf3f8274235d78bfa81f43019

## Possible Solution
Removing `sniper-tools` from from the include in `/settings.gradle.kts` allows the build to succeed.

## Steps to Reproduce
1. Run `./gradlew build`
2. For more information, see the logs in https://gist.github.com/kokx/ca709aebf3f8274235d78bfa81f43019

**Comment by kunyavskiy** [comment]
I've failed to reproduce it. Maybe the difference is in java 19 vs. java 17. 

Also I'm a bit confused with `error: release version 11 not supported`

**Comment by kokx** [comment]
Yes I'm confused by that error as well, because I do have java version 11 installed as well.

Anyway, I verified the same with java 17, and get the same errors: https://gist.github.com/kokx/b0a37ca5192438142d745dcf9f8a7579

**Comment by irdkwmnsb** [comment]
@kokx can you maybe check it now?

**Comment by kokx** [comment]
Just did. This unfortunately still gives the same errors. Building without `sniper-tools` still works however.

**Comment by kunyavskiy** [comment]
I still have no idea how to reproduce it. But I've converted it to koltin, and simplified the build.

Maybe some of this will help.

**Comment by irdkwmnsb** [comment]
@kokx can you check again?

**Comment by kokx** [comment]
Unfortunately this bug stays very illusive. This time I needed to build the software without `sniper-tools` and `reactions-bot` before the `StackOverflowError` would not occur.
